### PR TITLE
migrate mDNS to null safety

### PR DESCRIPTION
**Pull request by jonahwilliams**
Work towards https://github.com/flutter/flutter/issues/71511

### CODE REVIEW

File: packages/multicast_dns/lib/src/constants.dart [hunk 1]
```diff
@@ -14,20 +14,20 @@ final InternetAddress mDnsAddressIPv6 = InternetAddress('FF02::FB');
 const int mDnsPort = 5353;
 
 /// Enumeration of supported resource record class types.
-class ResourceRecordClass {
+abstract class ResourceRecordClass {
   // This class is intended to be used as a namespace, and should not be
   // extended directly.
-  factory ResourceRecordClass._() => null;
```
Comment (jonahwilliams) [inline packages/multicast_dns/lib/src/constants.dart:20]: This pattern is no long allowed.

File: packages/multicast_dns/pubspec.yaml [hunk 2]
```diff
@@ -1,14 +1,10 @@
 name: multicast_dns
 description: Dart package for mDNS queries (e.g. Bonjour, Avahi).
 homepage: https://github.com/flutter/packages/tree/master/packages/multicast_dns
-version: 0.2.2
-
-dependencies:
-  meta: ^1.1.6
```
Comment (jonahwilliams) [inline packages/multicast_dns/pubspec.yaml:7]: Meta was only used for required and for a protected annotation, but that annotation was used on private members and I don't think its worth the dependency.

File: packages/multicast_dns/lib/multicast_dns.dart [hunk 3]
```diff
@@ -48,14 +48,14 @@ class MDnsClient {
...
-  RawDatagramSocket _incoming;
   final List<RawDatagramSocket> _sockets = <RawDatagramSocket>[];
   final LookupResolver _resolver = LookupResolver();
   final ResourceRecordCache _cache = ResourceRecordCache();
   final RawDatagramSocketFactory _rawDatagramSocketFactory;
 
-  InternetAddress _mDnsAddress;
-  int _mDnsPort;
+  InternetAddress? _mDnsAddress;
+  late int _mDnsPort;
```
Comment (blasten) [inline packages/multicast_dns/lib/multicast_dns.dart:57]: The lookup of this value is protected by `_started`, which makes `late` ok, but what about making it nullable and explicitly checking for it before it's used.  I think it has the added benefit that removes the assumption that a separate bit keeps track of whether it's safe to read the value.

Comment (jonahwilliams) [reply packages/multicast_dns/lib/multicast_dns.dart]: done

File: packages/multicast_dns/lib/multicast_dns.dart [hunk 4]
```diff
@@ -48,14 +48,14 @@ class MDnsClient {
...
   final List<RawDatagramSocket> _sockets = <RawDatagramSocket>[];
   final LookupResolver _resolver = LookupResolver();
   final ResourceRecordCache _cache = ResourceRecordCache();
   final RawDatagramSocketFactory _rawDatagramSocketFactory;
 
-  InternetAddress _mDnsAddress;
-  int _mDnsPort;
+  InternetAddress? _mDnsAddress;
+  int? _mDnsPort;
+  late RawDatagramSocket _incoming;
```
Comment (blasten) [inline packages/multicast_dns/lib/multicast_dns.dart:58]: this one too :)

**Review by blasten — APPROVED** [review]

File: packages/multicast_dns/lib/multicast_dns.dart (hunk 4 above)
Comment (jonahwilliams) [reply packages/multicast_dns/lib/multicast_dns.dart:58]: fair enough :D

Comment (jonahwilliams) [reply packages/multicast_dns/lib/multicast_dns.dart]: didn't even need it

**Comment by miketheman** [comment]
Hooray, thanks for working on this. I looked on pub.dev and didn't see a release yet - is there a timeline or process I can follow to be notified of when this lands?

**Comment by stuartmorgan** [comment]
Looks like it just never got published; I've now done that.

**Pull request merged**